src/mapgen/mg_schematic.cpp

```diff
@@ -584,26 +606,28 @@ void Schematic::applyProbabilities(v3s16 p0,
 }
 
 
-void generate_nodelist_and_update_ids(MapNode *nodes, size_t nodecount,
-	std::vector<std::string> *usednodes, const NodeDefManager *ndef)
+void Schematic::condenseContentIds(size_t nodecount)
```

Comment:
is there any reason this would not be called on the entire volume?
If not I suggest removing this parameter.